File: kyw0716/level0/등수 매기기.js
function solution(score) {
  score = score.map((s) => {
    return (s[0] + s[1]) / 2;
  });

  let sorted = [...score];
  sorted.sort((a, b) => b - a);

  score = score.map((v) => {
    return sorted.indexOf(v) + 1;
  });

  return score;
}

// https://school.programmers.co.kr/learn/courses/30/lessons/120882

File: sk000801/입문/등수 매기기.java
import java.util.*;
class Solution {
    public int[] solution(int[][] score) {
        double[] scores = new double[score.length];
        double[] sort = new double[score.length];
        String[] arr = new String[score.length];
        int[] answer = new int[score.length];
        for(int i = 0; i < score.length; i++) {
            scores[i] = (double) (score[i][0]+score[i][1])/2; //혹시나 소수점때메
            sort[i] = scores[i];
        }
        Arrays.sort(sort);
        for(int i = 0; i < sort.length; i++) {
            arr[i] = String.valueOf(sort[sort.length-i-1]);
        }
        for(int i = 0; i < sort.length; i++) {
            answer[i] = Arrays.asList(arr).indexOf(String.valueOf(scores[i]))+1;
        }
        return answer;
    }
}